yksgi/src/lib.rs

```diff
@@ -0,0 +1,344 @@
+        // follow.
+        loop {
+            match self.pc.opcode() {
+                LLVMOpcode::LLVMAdd => self.add(),
+                LLVMOpcode::LLVMBitCast => self.bitcast(),
+                LLVMOpcode::LLVMBr => {
+                    self.branch();
+                    continue;
+                }
+                LLVMOpcode::LLVMCall => {
+                    self.call()
+                },
+                LLVMOpcode::LLVMGetElementPtr => self.gep(),
+                LLVMOpcode::LLVMLoad => self.load(),
+                LLVMOpcode::LLVMPHI => self.phi(),
+                LLVMOpcode::LLVMRet => {
+                    if self.ret() {
+                        return true;
+                    }
+                },
+                LLVMOpcode::LLVMStore => self.store(),
+                LLVMOpcode::LLVMSwitch => {
+                    self.switch();
+                    continue;
+                }
+                _ => todo!("{:?}", self.pc.as_str()),
+            }
+            self.pc = LLVMValue(LLVMGetNextInstruction(self.pc.0));
+        }
+    }
+
+    unsafe fn add(&mut self) {
+        let op1 = self.pc.get_operand(0);
+        let op2 = self.pc.get_operand(1);
+        let val1 = self.lookup(&op1);
+        let val2 = self.lookup(&op2);
+        // FIXME: Can overflows happen? What do we do?
+        let res = val1.val + val2.val;
+        self.setvar(self.pc, SGValue::new(res, self.pc.get_type()));
+    }
+
+    /// Convert value to new type of same bit size (without changing any bits).
+    pub unsafe fn bitcast(&mut self) {
+        // Since we store all integer values as u64 and evaluate them lazily according
+        // to the attached type, this operation is a NOP.
+        let src = LLVMValue(LLVMGetOperand(self.pc.0, 0));
+        let dstty = LLVMType(LLVMTypeOf(self.pc.0));
+        let val = self.lookup(&src);
+        self.setvar(self.pc, val.with_type(dstty));
+    }
+
+    /// Interpret branch instruction `instr`.
+    pub unsafe fn branch(&mut self) {
+        debug_assert!(self.pc.is_br());
+        let succ = if LLVMIsConditional(self.pc.0) == 0 {
+            LLVMBasicBlock::new(LLVMGetSuccessor(self.pc.0, 0))
+        } else {
+            let cond = LLVMValue::new(LLVMGetCondition(self.pc.0));
+            let val = self.lookup(&cond);
+            if val.val == 1 {
+                LLVMBasicBlock::new(LLVMGetSuccessor(self.pc.0, 0))
+            } else {
+                LLVMBasicBlock::new(LLVMGetSuccessor(self.pc.0, 1))
+            }
+        };
+        self.lastbb = Some(LLVMBasicBlock::new(LLVMGetInstructionParent(self.pc.0)));
+        self.pc = succ.first();
+    }
+
+    pub unsafe fn call(&mut self) {
+        // FIXME: Properly implement calls.
+        let func = LLVMGetCalledValue(self.pc.0);
+        let mut size = 0;
+        let name = CStr::from_ptr(LLVMGetValueName2(func, &mut size)).to_str().unwrap();
+        if name.starts_with("llvm.lifetime") {
```

Comment:
Why no behaviour for this?

Reply:
I think it's because it doesn't really do anything in the context of our interpreter:
https://llvm.org/docs/LangRef.html#int-lifestart

Reply:
OK, a quick comment explaining why would be helpful!